From 019dc0fe1ee21da8dbb6d0a9b3013210e355a68c Mon Sep 17 00:00:00 2001
From: Jingdong Lu <jingdong.lu@intel.com>
Date: Thu, 26 Oct 2023 10:40:10 +0800
Subject: [PATCH 22/54] Fix installer compiling error in ivshmem

Fix the following error while compiling installer:
error: cast to 'void *' from smaller integer type 'unsigned int'.

Tracked-On: OAM-112948
Signed-off-by: Jingdong Lu <jingdong.lu@intel.com>
---
 libkernelflinger/ivshmem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libkernelflinger/ivshmem.c b/libkernelflinger/ivshmem.c
index ab7b678..9df2819 100644
--- a/libkernelflinger/ivshmem.c
+++ b/libkernelflinger/ivshmem.c
@@ -406,7 +406,7 @@ EFI_STATUS ivshmem_init(void)
 
 	if (g_ivshmem_dev.revision == 1) {
 		info(L"IVSHMEM device: ivposition=%d",
-			io_read_32((void *)(g_ivshmem_dev.bar0_addr + IVPOSITION_OFF)));
+			io_read_32((void *)((UINT64)(g_ivshmem_dev.bar0_addr + IVPOSITION_OFF))));
 	}
 
 	return EFI_SUCCESS;
-- 
2.25.1

